src/editmap.cpp

```diff
@@ -1831,6 +1831,170 @@ int editmap::mapgen_preview( real_coords &tc, uimenu &gmenu )
+            std::memcpy( destsm->lum, srcsm->lum, sizeof( srcsm->lum ) ); // emissive items
+
+
+            for( int sx = 0; sx < SEEX; ++sx ) {
+                for( int sy = 0; sy < SEEY; ++sy ) {
+                    for( auto &elem : srcsm->itm[sx][sy] ) {
+                        destsm->itm[sx][sy].push_back(elem);
+                    }
+                    //Don't cover existing crops, for farm upgrades
+                    //Don't destroy terrain to place grass/dirt
+                    if( change_sensitive && destsm->frn[sx][sy] != furn_str_id( "f_plant_seed" ) &&
+                        destsm->frn[sx][sy] != furn_str_id( "f_plant_seedling" ) &&
+                        destsm->frn[sx][sy] != furn_str_id( "f_plant_mature" ) &&
+                        destsm->frn[sx][sy] != furn_str_id( "f_plant_harvest" ) &&
+                        srcsm->ter[sx][sy] != ter_str_id( "t_grass" ) &&
+                        srcsm->ter[sx][sy] != ter_str_id( "t_dirt" ) ) {
+                               destsm->ter[sx][sy] = srcsm->ter[sx][sy];
+                               destsm->frn[sx][sy] = srcsm->frn[sx][sy];
+                       }
+                    destsm->fld[sx][sy] = srcsm->fld[sx][sy];
+                    destsm->cosmetics[sx][sy].swap( srcsm->cosmetics[sx][sy] );
+                }
+            }
+
+            // various misc variables
+            destsm->active_items = srcsm->active_items;
+
+            destsm->temperature = srcsm->temperature;
+            destsm->last_touched = calendar::turn;
+            destsm->comp = std::move( srcsm->comp );
+            destsm->camp = srcsm->camp;
+
+            if( spawns_todo > 0 ) {
+                g->m.spawn_monsters( true );
+            }
+        }
+    }
+    g->m.reset_vehicle_cache( target.z );
+
+    cleartmpmap( tmpmap );
+    return true;
+}
+
+vehicle* editmap::mapgen_veh_query( tripoint omt_tgt, vehicle* car_target, bool destroy)
+{
+    tinymap target_bay;
+    target_bay.load( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, false );
+
+    std::vector<vehicle *> possible_vehicles;
+    std::vector<vehicle *> sub_possible_vehicles;
+    for( int x = 0; x < 2; x++ ) {
+        for( int y = 0; y < 2; y++ ) {
+            submap *destsm = target_bay.get_submap_at_grid( x, y, target.z );
+            int target_pos = -1;
+            for( size_t z = 0; z < destsm->vehicles.size(); z++ ){
+                possible_vehicles.push_back(destsm->vehicles[z]);
+                if( destsm->vehicles[z] == car_target && destroy){
+                    target_pos = z;
+                    auto &ch = g->m.access_cache( destsm->vehicles[z]->smz );
+                    ch.vehicle_list.erase( destsm->vehicles[z] );
+                }
+            }
+            if( destroy && target_pos != -1 ){
+                destsm->vehicles.erase(destsm->vehicles.begin() + target_pos);
+                g->m.update_vehicle_list( destsm, target.z );
```

Comment:
You're updating vehicle list of current map rather than `target_bay`. This can't work.